### PR TITLE
Convert path variable to char*

### PR DESCRIPTION
**Pull request by anttiylitokola**

### CODE REVIEW

**Comment by anttiylitokola** [comment]
@TeroJaasko

File: nsdl-c/sn_nsdl_lib.h [hunk 1]
```diff
@@ -437,51 +435,47 @@ extern int8_t sn_nsdl_put_resource(struct nsdl_s *handle, sn_nsdl_dynamic_resour
 extern int8_t sn_nsdl_pop_resource(struct nsdl_s *handle, sn_nsdl_dynamic_resource_parameters_s *res);
 
 /**
- * \fn extern int8_t sn_nsdl_delete_resource(struct nsdl_s *handle, uint8_t pathlen, uint8_t *path)
+ * \fn extern int8_t sn_nsdl_delete_resource(struct nsdl_s *handle, char *path)
```
Comment (TeroJaasko) [inline nsdl-c/sn_nsdl_lib.h]: This path argument here and most other places could/should now be "const char*".

File: source/libNsdl/src/sn_grs.c [hunk 2]
```diff
@@ -491,13 +482,12 @@ int8_t sn_grs_put_resource(struct grs_s *handle, sn_nsdl_dynamic_resource_parame
     }
 
     /* Check path validity */
-    if (!res->static_resource_parameters->pathlen || !res->static_resource_parameters->path) {
+    if (!res->static_resource_parameters->path) {
```
Comment (TeroJaasko) [inline source/libNsdl/src/sn_grs.c]: Original version checked also that path's length is >0, should that functionality be maintained here? One could check that eg. by adding a (res->static_resource_parameters->path[0] == '\0') check.

File: source/libNsdl/src/sn_grs.c [hunk 3]
```diff
@@ -517,13 +507,12 @@ int8_t sn_grs_pop_resource(struct grs_s *handle, sn_nsdl_dynamic_resource_parame
...
```
Comment (TeroJaasko) [inline source/libNsdl/src/sn_grs.c]: Same comment about zero length check here too.

File: source/libNsdl/src/sn_grs.c [hunk 4]
```diff
@@ -887,7 +875,14 @@ sn_nsdl_dynamic_resource_parameters_s *sn_grs_search_resource(struct grs_s *hand
         /* Scan all nodes on list */
         ns_list_foreach(sn_nsdl_dynamic_resource_parameters_s, resource_search_temp, &handle->resource_root_list) {
             /* If length equals.. */
-            if (resource_search_temp->static_resource_parameters->pathlen == pathlen) {
+            size_t len = 0;
+            if(resource_search_temp &&
+                    resource_search_temp->static_resource_parameters &&
+                    resource_search_temp->static_resource_parameters->path) {
```
Comment (TeroJaasko) [inline source/libNsdl/src/sn_grs.c]: Are those "resource_search_temp && resource_search_temp->static_resource_parameters" checks really needed? The resource_search_temp should never be null here if I remember ns_list_foreach correctly, and the static_resource_parameters should be validated when a new item is added to list, not afterwards.

File: source/libNsdl/src/sn_nsdl.c [hunk 5]
```diff
@@ -909,10 +909,14 @@ int8_t sn_nsdl_build_registration_body(struct nsdl_s *handle, sn_coap_hdr_s *mes
 
             *temp_ptr++ = '<';
             *temp_ptr++ = '/';
+            size_t path_len = 0;
+            if (resource_temp_ptr->static_resource_parameters->path) {
```
Comment (TeroJaasko) [inline source/libNsdl/src/sn_nsdl.c]: Is this path != null check needed here? On what cases can it be NULL? If it was verified when item was added to list, this and few other places could be simplified a bit.

**Comment by TeroJaasko** [comment]
+1 for change itself. Please consider the comments and apply those which are valid and which you agree with:)

**Pull request merged**